reporting-cli/docs/user/Reference Manual.md

```diff
@@ -0,0 +1,131 @@
+## Reporting CLI
+
+Reporting CLI is a quick out-of-box options to be able to download reports without requiring the use of Dashboards.
+
+### Features
+
+- Reporting CLI supports following authentication types for opensearch connection.
+    - Basic authentication
+    - Cognito authentication
+    - SAML authentication
+    - No auth
+- Reporting CLI supports downloading reports in below file formats.
+    - PDF
+    - PNG
+    - CSV
+- Reporting CLI can be used to send email with report as an attachment.   
+
+### Command-line Options
+
+Reporting CLI supports following configurable options.
+
+Option | Default Value | Valid Options |  Environment Variable | Description
+-- | --- | --- | --- | --- |
+url | - | - | URL | url of the report
+format | pdf | pdf, png, csv | - | file formart of the report
+width | 1680 | - | - | page width in pixels
+height | 600 | - | - | page height in pixels
+filename | reporting | - | FILENAME | file name of the report
+auth | none | basic, saml, cognito, none | - | authentication type
+credentials | - | - | USERNAME and PASSWORD | login credentials in the format of username:password for connecting to url
+from | - | - | FROM | the email address of the sender
+to | - | - | TO | the recipient of the email
+transport | - | ses, smtp | TRANSPORT | transport for sending the email
+smtphost | - | - | SMTP_HOST | the hostname of the SMTP server
+smtpport | - | - | SMTP_PORT | the port for connection
+smtpusername | - | - | SMTP_USERNAME | SMTP username
+smtppassword | - | - | SMTP_PASSWORD | SMTP password
+smtpsecure | - | - | SMTP_SECURE | if true the connection will use TLS when connecting to server.
+subject | 'This is an email containing your dashboard report' | - | SUBJECT |Subject for the email
+
+You can also find this information using help command.
+```
+reporting --help
+```
+
+### Environment Variable File
+
+Reporting CLI also reads environment variables from .env file inside the project. 
+
+```md
+.
+Reporting
+├── src
+│   └── .env
+```
+
+- Each line should have format NAME=VALUE
+- Lines starting with # are considered as comments.
+- There is no special handling of quotation marks.
+
+NOTE: Values from the environment file has higher priority than command line argument. For example, if you add filename as *test* in *.env* file and also add `--filename report` command option, the downloded report's name will be *test*.
```

Comment:
just an opinion but option is more explicit than process.env, should options overwrite env?

Reply:
that make sense. I updated the code and to use option first.